src/libopensc/card-epass2003.c

```diff
@@ -2440,12 +2453,42 @@ epass2003_gen_key(struct sc_card *card, sc_epass2003_gen_key_data * data)
 	if (len < apdu.resplen)
 		LOG_FUNC_RETURN(card->ctx, SC_ERROR_INVALID_ARGUMENTS);
 
+	if(rbuf == NULL)
+		LOG_FUNC_RETURN(card->ctx, SC_ERROR_INVALID_ARGUMENTS);
+		
```

Comment:
See the failure in the clang build:
```
card-epass2003.c:2463:5: error: comparison of array 'rbuf' equal to a null pointer is always false [-Werror,-Wtautological-pointer-compare]

        if(rbuf == NULL)

           ^~~~    ~~~~

1 error generated.
```
This condition is completely bogus.

Reply:
Yes, already found it, and R&D working on it, back to you very soon.

Reply:
fixed, have a check, thanks